src/sentry/runner/commands/run.py

```diff
@@ -557,17 +557,36 @@ def metrics_consumer(**options):
 @click.option("commit_max_batch_size", "--commit-max-batch-size", type=int, default=25000)
 @click.option("commit_max_batch_time", "--commit-max-batch-time-ms", type=int, default=10000)
 def metrics_streaming_consumer(**options):
+    from arroyo.backends.kafka import KafkaProducer
+    from confluent_kafka import Producer
+    from django.conf import settings
+
     from sentry.sentry_metrics.metrics_wrapper import MetricsWrapper
     from sentry.sentry_metrics.multiprocess import get_streaming_metrics_consumer
+    from sentry.utils import kafka_config
     from sentry.utils.metrics import backend
 
+    snuba_metrics = settings.KAFKA_TOPICS[settings.KAFKA_SNUBA_METRICS]
+    if options.get("factory_name") == "multiprocess":
+        snuba_metrics_producer = KafkaProducer(
+            kafka_config.get_kafka_producer_cluster_options(snuba_metrics["cluster"]),
+        )
+    else:
+        snuba_metrics_producer = Producer(
+            kafka_config.get_kafka_producer_cluster_options(snuba_metrics["cluster"]),
+        )
+    producer = snuba_metrics_producer
+    options["producer"] = producer
+
     metrics = MetricsWrapper(backend, "sentry_metrics.indexer")
     configure_metrics(metrics)
 
     streamer = get_streaming_metrics_consumer(**options)
 
     def handler(signum, frame):
         streamer.signal_shutdown()
+        if options.get("factory_name") == "multiprocess":
+            producer.close()
```

Comment:
I dont think i can do 

```
with closing(producer):
    streamer.run()
```

because the confluent producer doesn't have a `close` method, that's something we add onto the arroyo producer